common/src/main/java/com/wynntils/handlers/scoreboard/ScoreboardHandler.java

```diff
@@ -4,346 +4,361 @@
-
-    private final List<Pair<ScoreboardPart, Set<SegmentMatcher>>> scoreboardParts = new ArrayList<>();
-    private List<SegmentMatcher> segmentMatchers = List.of();
-
-    private ScheduledExecutorService executor = null;
+    public void addPart(ScoreboardPart scoreboardPart) {
+        scoreboardParts.add(scoreboardPart);
+    }
 
-    private boolean firstExecution = false;
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onSetScore(ScoreboardSetScoreEvent event) {
+        if (!Objects.equals(event.getObjectiveName(), "sb" + McUtils.player().getScoreboardName())) {
+            return;
+        }
 
-    private void periodicTask() {
-        // FIXME: Reverse dependency
-        if (!Models.WorldState.onWorld() || McUtils.player() == null) return;
+        handleSetScore(event.getOwner(), event.getScore(), event.getMethod());
+    }
 
-        if (queuedChanges.isEmpty()) {
-            handleScoreboardReconstruction();
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onSetObjective(ScoreboardSetObjectiveEvent event) {
+        if (!Objects.equals(event.getObjectiveName(), "sb" + McUtils.player().getScoreboardName())) {
             return;
         }
 
-        List<ScoreboardLine> scoreboardCopy = new ArrayList<>(reconstructedScoreboard);
-        LinkedList<ScoreboardLineChange> queueCopy = new LinkedList<>(queuedChanges);
-        queuedChanges.clear();
+        if (event.getMethod() != ScoreboardSetObjectiveEvent.METHOD_REMOVE) return;
 
-        Map<Integer, ScoreboardLine> scoreboardLineMap = new TreeMap<>();
-
-        for (ScoreboardLine scoreboardLine : scoreboardCopy) {
-            scoreboardLineMap.put(scoreboardLine.index(), scoreboardLine);
-        }
+        // Reset the scoreboard
+        reconstructedScoreboard.clear();
 
-        Set<String> changedLines = new HashSet<>();
-        while (!queueCopy.isEmpty()) {
-            ScoreboardLineChange processed = queueCopy.pop();
+        // Reset the scoreboard segments
+        calculateScoreboardSegments();
+        createScoreboardFromSegments();
+    }
 
-            if (processed.method() == ServerScoreboard.Method.REMOVE) {
-                for (ScoreboardLine lineToRemove : scoreboardLineMap.values().stream()
-                        .filter(scoreboardLine -> Objects.equals(scoreboardLine.line(), processed.lineText()))
-                        .toList()) {
-                    scoreboardLineMap.remove(lineToRemove.index());
-                }
-            } else {
-                ScoreboardLine line = new ScoreboardLine(processed.lineText(), processed.lineIndex());
-                scoreboardLineMap.put(line.index(), line);
-                changedLines.add(processed.lineText());
-            }
-        }
+    @SubscribeEvent
+    public void onSetObjectiveDisplay(ScoreboardSetDisplayObjectiveEvent event) {
+        event.setCanceled(true);
```

Comment:
I'm still a bit lost at the minecraft scoreboard packets. What that this poor one do, that causes it to be always canceled?

Reply:
This sets the scoreboard to be seen on the screen. But we want to display ours. So we have to ignore Wynn.

Reply:
Ah, I see. I think I see, at least. For those scoreboard parts that are not handled by others, do we render them ourselves? I thought we modified the scoreboard but still let Vanilla render it, but I might be misunderstanding what I'm seeing.

Reply:
We can render any part without processing. The only processing we do is setting visibility. If no visibility was changed, we basically clone the wynn scoreboard, and display it.

Reply:
And we basically let vanilla render it, we just create a different display.

Reply:
Ok, I think I get it now. Mostly. :) I think my main problem here is that I don't fully understand the vanilla mechanics, since I've only seen how Wynncraft mis-uses it. But I'm satisfied with my current level of understanding. :)